scene/2d/camera_2d.cpp

```diff
@@ -562,32 +535,46 @@ real_t Camera2D::get_drag_vertical_offset() const {
 void Camera2D::set_drag_horizontal_offset(real_t p_offset) {
 	drag_horizontal_offset = p_offset;
 	drag_horizontal_offset_changed = true;
-	Point2 old_smoothed_camera_pos = smoothed_camera_pos;
-	_update_scroll();
-	smoothed_camera_pos = old_smoothed_camera_pos;
+	_update_position();
 }
 
 real_t Camera2D::get_drag_horizontal_offset() const {
 	return drag_horizontal_offset;
 }
 
-void Camera2D::_set_old_smoothing(real_t p_enable) {
-	//compatibility
-	if (p_enable > 0) {
-		smoothing_enabled = true;
-		set_follow_smoothing(p_enable);
+void Camera2D::set_enable_follow_smoothing(bool p_enabled) {
+	if (smoothing_enabled == p_enabled) {
+		return;
 	}
-}
 
-void Camera2D::set_enable_follow_smoothing(bool p_enabled) {
+	// Separate the logic between enabled and active, because the smoothing
+	// cannot be active in the editor. This can be done without a separate flag
+	// but is bug prone so this approach is easier to follow.
```

Comment:
I think you could just add an inline method `is_smoothing_active()`.

Reply:
That's fine too. (That was ported from my PR actually lol :grin: ).

Reply:
Yes, this is from the forward port of #46717.